torch/_export/logical_schema.py

```diff
@@ -206,16 +206,21 @@ class Tensor:
 
 
 ################################################################################
-# Following section is the defining the schema of 3 level construct: GraphModule, Graph, Node
+# Following section is defining the schema of 3 level construct: GraphModule, Graph, Node
 
-# IValue has no corresponding class in fx
-# IValue is the "values" that are passed between nodes in the graph
-# IValue is a named virtual tensor, with an optional TensorMeta that describes the properties of the tensor
-# !!! Consider using a more descriptive name, e.g. TensorValue, TensorPlaceholder, TensorArgument, etc.
+# TensorValue has no corresponding class in fx
+# TensorValue is the "tensor results" that are passed between nodes in the graph
+# TensorValue is a named virtual tensor, with an optional TensorMeta that describes the properties of the tensor
 @dataclass
-class IValue:
-    meta: TensorMeta
+class TensorValue:
+    name: str                   # unique identifier of the TensorValue, referenced in Argument.as_tensor field
+    meta: Optional[TensorMeta]  # optional tensor meta
```

Comment:
when would this be None?

Reply:
In our design, TensorValue should have metadata, but I am not sure if we can always uphold this promise. Especially, when there is custom op that doesn't have meta function.

Reply:
For custom ops we plan to add a requirement that a metafunction be defined (otherwise exporting breaks in all sorts of other ways). So if that's the only reason for this field to be optional, we should change it.

Reply:
OK. I am removing the Optional annotation.